src/geom/reference_elem.C

```diff
@@ -247,11 +247,20 @@ const Elem & get (const ElemType Type)
 {
   libmesh_assert_less (Type, INVALID_ELEM);
 
+  ElemType BaseType = Type ;
+
+  // For shell elements, use non shell type as the base type
+  if (Type == TRISHELL3)
+    BaseType = TRI3 ;
+
+  if (Type == QUADSHELL4)
+    BaseType = QUAD4 ;
+
   init_ref_elem_table();
 
-  libmesh_assert (ref_elem_map[Type] != libmesh_nullptr);
+  libmesh_assert (ref_elem_map[BaseType] != libmesh_nullptr);
 
-  return *ref_elem_map[Type];
+  return *ref_elem_map[BaseType];
 }
```

Comment:
This is probably OK, other than the spurious spaces before semi-colons.

I think the long term fix here is to generate new entries in the reference_elem.data file for the new element types.  These entries include boundary conditions, so they would be slightly different than the existing QUAD4 and TRI3 types.  However, I think these are generated from the files in the reference_elements/{2D,3D} directories, and we don't yet support writing the new shell types to XDA files... so that would need to be fixed first.